pkg/daemon/ceph/client/pool.go

```diff
@@ -154,27 +154,29 @@ func ParsePoolDetails(in []byte) (CephStoragePoolDetails, error) {
 	return poolDetails, nil
 }
 
-func CreatePoolWithProfile(context *clusterd.Context, clusterInfo *ClusterInfo, clusterSpec *cephv1.ClusterSpec, poolName string, pool cephv1.PoolSpec, appName string) error {
+func CreatePoolWithProfile(context *clusterd.Context, clusterInfo *ClusterInfo, clusterSpec *cephv1.ClusterSpec, pool cephv1.NamedPoolSpec, appName string) error {
+	if pool.Name == "" {
```

Comment:
How can we end up in this condition?

Reply:
We shouldn't end up here, it is a check to make sure we don't

Reply:
But the Kubernetes admission controller verifies this for us already right?

Reply:
Agreed at the highest level it will be checked there. With so many places creating pools, this was really a check to make sure the unit tests were setting it.

Reply:
Actually to be more correct, this `Name` is an optional field in the pool spec, but the controllers are expected to set it before calling the ceph client API.